TOC.md

```diff
@@ -2,187 +2,204 @@
+      - [Data Manipulation Statements](sql/dml.md)
+      - [Transactions](sql/transaction.md)
+      - [Database Administration Statements](sql/admin.md)
+      - [Prepared SQL Statement Syntax](sql/prepare.md)
+      - [Utility Statements](sql/util.md)
+      - [TiDB SQL Syntax Diagram](https://pingcap.github.io/sqlgram/)
+    + Data Types
+      - [Numeric Types](sql/datatype.md#numeric-types)
+      - [Date and Time Types](sql/datatype.md#date-and-time-types)
+      - [String Types](sql/datatype.md#string-types)
+      - [JSON Types](sql/datatype.md#json-types)
+      - [The ENUM data type](sql/datatype.md#the-enum-data-type)
+      - [The SET Type](sql/datatype.md#the-set-type)
+      - [Data Type Default Values](sql/datatype.md#data-type-default-values)
+    - [Constraints](sql/constraints.md)
+    + Functions and Operators
+      - [Function and Operator Reference](sql/functions-and-operators-reference.md)
+      - [Type Conversion in Expression Evaluation](sql/type-conversion-in-expression-evaluation.md)
+      - [Operators](sql/operators.md)
+      - [Control Flow Functions](sql/control-flow-functions.md)
+      - [String Functions](sql/string-functions.md)
+      - [Numeric Functions and Operators](sql/numeric-functions-and-operators.md)
+      - [Date and Time Functions](sql/date-and-time-functions.md)
+      - [Bit Functions and Operators](sql/bit-functions-and-operators.md)
+      - [Cast Functions and Operators](sql/cast-functions-and-operators.md)
+      - [Encryption and Compression Functions](sql/encryption-and-compression-functions.md)
+      - [Information Functions](sql/information-functions.md)
+      - [JSON Functions](sql/json-functions.md)
+      - [Aggregate (GROUP BY) Functions](sql/aggregate-group-by-functions.md)
+      - [Miscellaneous Functions](sql/miscellaneous-functions.md)
+      - [Precision Math](sql/precision-math.md)
+    + SQL Language Structure
+      - [Literal Values](sql/literal-values.md)
+      - [Schema Object Names](sql/schema-object-names.md)
+      - [Keywords and Reserved Words](sql/keywords-and-reserved-words.md)
+      - [User-Defined Variables](sql/user-defined-variables.md)
+      - [Expression Syntax](sql/expression-syntax.md)
+      - [Comment Syntax](sql/comment-syntax.md)
+    - [Generated Columns](sql/generated-columns.md)
+    + Transactions
+      - [Transaction Model](sql/transaction-model.md)
+      - [Isolation Levels](sql/transaction-isolation.md)
+    + Performance
+      - [Overview](sql/sql-optimizer-overview.md)
+      - [Cost-based Optimizer](sql/understanding-the-query-execution-plan.md)
+      - [Introduction to Statistics](sql/statistics.md)
+      - [Optimizer Hints](sql/optimizer-hints.md)
+      - [Tune TiKV](op-guide/tune-tikv.md)
+      - [TiDB Best Practices](https://pingcap.com/blog/2017-07-24-tidbbestpractice/)
+    - [The mysql System Database](sql/system-database.md)
+    - [Information Schema](sql/information-schema.md)
+    + TiSpark
```

Comment:
It might be inappropriate to put TiSpark under SQL. Perhaps we can put it under “Deploy” and think about the exact wording.